### PR TITLE
Parse campaign timezone offsets

### PR DESCRIPTION
**Pull request by crayolakat**
## Description

If there are campaign timezone offsets (e.g.: in the test-fakedata contact loader), process these offsets and determine whether the offsets are valid or invalid timezones. Without this code, contacts with a timezone offset that isn't defined in the `ALL_OFFSETS` constant aren't processed and won't be textable.

# Checklist:

- [ ] I have manually tested my changes on desktop and mobile
- [X] The test suite passes locally with my changes
- [ ] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [X] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- [ ] I have made any necessary changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] My PR is labeled [WIP] if it is in progress

### CODE REVIEW

**Review by schuyler1d — DISMISSED** [review]
:100: awesome to do this -- would suggest writing a test -- the more timezone tests the better because that stuff is hard!

**Comment by crayolakat** [comment]
After Sky's review, I did the following:

1. Merged in the changes from [this PR](https://github.com/MoveOnOrg/Spoke/pull/2090) to fix GitHub Actions automated tests failing
2. Updated code to make campaign offsets unique and wrote test ([commit](https://github.com/MoveOnOrg/Spoke/pull/2067/commits/eb9a87c54486c3c9da4957ecd56216d21448ec26))

File: src/components/AssignmentTexter/ContactController.jsx [hunk 1]
```diff
@@ -187,9 +187,11 @@ export class ContactController extends React.Component {
       // console.log('getContactData length', newIndex, getIds.length)
       this.setState({ loading: true });
       const contactData = await this.props.loadContacts(getIds);
-      const {
-        data: { getAssignmentContacts }
-      } = contactData;
+      let getAssignmentContacts;
+      if (contactData && contactData.data) {
+        getAssignmentContacts = contactData.data.getAssignmentContacts;
```
Comment (ibrand) [inline src/components/AssignmentTexter/ContactController.jsx:192]: This seems unrelated to most of the other changes, what's this in here for? Is it related to a bug?

**Review by ibrand — APPROVED** [review]
One question, but otherwise looks good!

File: src/components/AssignmentTexter/ContactController.jsx (hunk 1 above)
Comment (crayolakat) [reply src/components/AssignmentTexter/ContactController.jsx:192]: This change is from the PR that fixed the GitHub Actions failing automated tests: https://github.com/MoveOnOrg/Spoke/pull/2090

I cherry-picked the changes from that PR into this PR so that the automated tests wouldn't all fail.
